source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -242,11 +273,15 @@ void AuthenticatorImpl::onDestroy() {
 void AuthenticatorImpl::verifyKey() {
   const Status status =
       ::google::jwt_verify::verifyJwtWithoutTimeChecking(*jwt_, *jwks_data_->getJwksObj());
+
   if (status != Status::Ok) {
     doneWithStatus(status);
     return;
   }
+  handleGoodJwt();
+}
 
+void AuthenticatorImpl::handleGoodJwt() {
```

Comment:
add function argument in   handleGoodJwt(bool cache_hit) 

inside the function,  before line 304, add

```
  if (provider_) {
      // move the ownership of "new_jwt_" into the function.
      jwks_data_->getJwtCache().add(curr_token_->token(), std::move(new_jwt_));
  }

Reply:
Done.